autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -936,26 +942,50 @@ def get_models_load_info(self, model_names):
         model_types = {model_name: self.model_types[model_name] for model_name in model_names}
         return model_names, model_paths, model_types
 
-    # TODO: Add pred_time_val_full (Will be incorrect unless graph representation is added)
+    # Sums the attribute value across all models that the provided model depends on, including itself.
+    # For instance, this function can return the expected total predict_time of a model.
+    # attribute is the name of the desired attribute to be summed.
+    def get_model_attribute_full(self, model, attribute):
+        base_model_set = self.get_minimum_model_set(model)
+        if len(base_model_set) == 1:
+            return self.model_graph.nodes[base_model_set[0]][attribute]
+        attribute_full = 0
+        for base_model in base_model_set:
+            attribute_full += self.model_graph.nodes[base_model][attribute]
+        return attribute_full
+
+    # Gets the minimum set of models that the provided model depends on, including itself
+    # Returns a list of model names
+    def get_minimum_model_set(self, model):
+        if not isinstance(model, str):
+            model = model.name
+        return list(nx.bfs_tree(self.model_graph, model, reverse=True))
+
     def leaderboard(self):
         model_names = self.get_model_names_all()
         score_val = []
         fit_time = []
         pred_time_val = []
         stack_level = []
+        fit_time_full = []
+        pred_time_val_full = []
         for model_name in model_names:
             score_val.append(self.model_performance.get(model_name))
             fit_time.append(self.model_fit_times.get(model_name))
+            fit_time_full.append(self.get_model_attribute_full(model=model_name, attribute='fit_time'))
             pred_time_val.append(self.model_pred_times.get(model_name))
+            pred_time_val_full.append(self.get_model_attribute_full(model=model_name, attribute='predict_time'))
             stack_level.append(self.get_model_level(model_name))
         df = pd.DataFrame(data={
             'model': model_names,
             'score_val': score_val,
+            'fit_time_full': fit_time_full,
+            'pred_time_val_full': pred_time_val_full,
             'fit_time': fit_time,
             'pred_time_val': pred_time_val,
```

Comment:
I'm a bit worried users will be confused by these names, so they should be clearly described in the Predictor.leaderboard() documentation.

In just this leaderboard df, perhaps it's worth adding a suffix to : `fit_time`, `pred_time_val` to contrast them against the `_full` suffix?  Options for the suffix might include: `_base`, `_part`, `_extra`, `_solo`, `_marginal`,...

Reply:
Agreed, wasn't sure at the time how to name them.

Added descriptions in documentation.

Changed _full to no suffix and the previous one to _marginal